### DISCUSSION TITLE
command for default font and default colour?

### DISCUSSION THREAD

**Discussion by JacobBeaudry** [KaTeX/KaTeX#2820] [Q&A]
Hello!

I was wondering if there is any programmatic way to escape font or colour? It seems that  the `\color{normalcolor}` switch does not behave like other colour switches in that it is overrun by any other colour command (compare the first and second lines of the following screenshot, where the first line uses `\color{normalcolor}` whereas the 2nd uses `black`):
![image (3)](https://user-images.githubusercontent.com/75229506/109838451-cb2adb00-7c0b-11eb-935e-423223236b1a.png)

Unfortunately `black` is slightly more emphasized than the default colour, and so creates a noticeable difference in the output. 

A similar issue exists for `mathnormal`: as [this question](https://tex.stackexchange.com/questions/555081/why-do-numbers-in-mathnormal-not-match-numbers-without-it) on the tex.se explains, numbers printed using `mathnormal` don't actually print in the "normal" (_i.e._ default) math font. Numbers will be correctly rendered using `mathrm`; however, `\mathrm` affects rendering of alphabetic variables. As such, there is no way of escaping a change of font in such a way that both numbers and letters are rendered in the default font: 
![image](https://user-images.githubusercontent.com/75229506/109839425-b4d14f00-7c0c-11eb-994e-d266bfa11cc5.png)

I am working for an NPO as a content creator, and have effectively no technical background in programming; however, if I can find some method for solving this (programmatic or otherwise), then I can pass it along to our chief programmer for him to implement. I've been wrestling with this particular issue for several months now, so would greatly appreciate any suggestions whatsoever for possible strategies to resolve the issue. 

Thank you very much, in advance, for your time and help, and thank you so much for the incredible work done, so far. I hope our team will be able to contribute to this project, moving forward.

All the best, 

Jacob

**Comment by JacobBeaudry** [comment]
Was really hoping for a reply on this. I started using `\htmlStyle` to set font-weight and color to default; however, `\htmlStyle{color:<color>}` never takes precedence over the built-in `\color` command e.g. 
```
{\color{red} \htmlStyle{color:black}{x}}
```
and
```
\htmlStyle{color:black}{{\color{red} x}}
```
both print a red x; moreover, `\htmlStyle{color:red}` does not work like a switch, the same way `\color{red}` does, so there is still no way to have escapable color & font styling.

**Comment by edemaine** [comment]
`\color` (and `\textcolor`) result in inline style elements.  You should be able to permanently escape color via `\htmlClass{escape-color}{...}` and using the following CSS:
```css
.escape-color, .escape-color * { color: normalcolor !important }
```
But you wouldn't be able to set color again within such an escaped area. Is this useful / what you want?

To escape font settings, it seems like the best way is to switch to text mode and back again, e.g.:
```latex
\text{$abc123$}
```
I've tested this in `\bf` and `\it` contexts, and it seems to treat letters and numbers like the initial state.

**Reply by JacobBeaudry** [reply]
> \color (and \textcolor) result in inline style elements. You should be able to permanently escape color via \htmlClass{escape-color}{...} and using the following CSS:

> .escape-color, .escape-color * { color: normalcolor !important }
> But you wouldn't be able to set color again within such an escaped area. Is this useful / what you want?

Unfortunately, no: I'm trying to set up some macros so that they can be easily coloured, without applying the colour to their arguments, while still leaving the possibility of styling the arguments themselves with another color. 

For now, I've reimplemented all colours using htmlStyle. It introduced a few other challenges, but so far it seems to be much more workable than the native TeX options. I also opted to do this for bold, as well.

Thanks for the suggestions -- they are useful tools to be aware of. Really appreciate your work & willingness to help and share your knowledge!

**Reply by edemaine** [reply]
Looking at this again, I realize that `\color{normalcolor}` will reset to the initial color as you wanted (at least in current KaTeX, but not LaTeX), because the colors are just translated to `style` settings.

Example, with a custom background sheet that changes the colors:

![image](https://user-images.githubusercontent.com/2218736/134979274-7381b09f-a416-44f2-b7aa-f92b369ed295.png)

**Reply by JacobBeaudry** [reply]
That doesn't actually work on all elements, _e.g._:  
![image](https://user-images.githubusercontent.com/75229506/134991876-e11079b9-42af-4a3e-95c6-f33b82b00b3a.png)

This is what threw me off, personally: I did all my tests on simple elements & then discovered that nothing worked when I was inside a fraction or similar environments. 

Also, FYI the solution of escaping to text mode to reset the font is _too_ effective and led to undesirable results when used, as it escapes the font-size as well. _E.g._ in footnote / substack / over|underbrace etc. it returns the result in the default font-size, so it's just trading one problem for another 😄
